### ISSUE TITLE
Model updates R31 register for linking only when the branch is taken

### ISSUE THREAD

**Issue by guptasanchi** [SimpleCPU/SimpleCPU#23]
For the BLTZAL instruction, I see the following mismatch - 
```
# PC:0000048c	INSTR:0690ff9a	 BLTZAL R20, 65434
# [RTL]  	PC:0000048c	Instr:0690ff9a	R20:00000000	R31:00000490
# [MODEL]	PC:0000048c	Instr:0690ff9a	R20:00000000	R31:000000a4
# 
# RTL R31 VAL: 490	 Model R31 VAL: a4
# RT Value Mismatch
# ** Fatal: TEST FAILED
```
There seems to be a mismatch between the R31 values reported by the RTL and the model. The RTL reports the value as CURRENT_PC + 4 but model doesn't update the link register. 
On further analysis, it seems that the branch is not taken and model updates the register only if the branch is taken. This doesn't seem to match with the MIPSISA specifications - 

> Unconditionally, the address of the instruction after the delay slot is placed in the link register, r31.

Model should update the link register even on not-taken branches.

**Comment by raulbehl** [comment]
Looks like a bug. Looking into it.

**Comment by raulbehl** [comment]
Fix in 10e5e11
